### ISSUE TITLE
Make available for download JRE too for Corretto 11 

### ISSUE THREAD

**Issue by Constantin07** [corretto/corretto-11#176]
### Is your feature request related to a problem?
I want to use minimal JRE when deploying application in docker container, not fat JDK.

### Describe a solution you would like
Be able to download only JRE.

### Describe alternatives you have considered
Using JDK but that's not optimal for deployments.

### Additional context
None.

**Comment by ghost** [comment]
@Constantin07 Thank you for the feature request. We will review it and get back to you.

**Comment by andrei-bamboi** [comment]
We want to use an JRE version , instead of JDK .

**Comment by eastig** [comment]
Hi @Constantin07, @andrei-bamboi 
For what OSes do you need JREs?
As a workaround you can use `jlink`: https://docs.oracle.com/en/java/javase/11/tools/jlink.html
To get the list of all modules in JDK:
```
$ java --list-modules
java.base@11.0.10
java.compiler@11.0.10
java.datatransfer@11.0.10
java.desktop@11.0.10
java.instrument@11.0.10
java.logging@11.0.10
java.management@11.0.10
java.management.rmi@11.0.10
java.naming@11.0.10
java.net.http@11.0.10
java.prefs@11.0.10
java.rmi@11.0.10
java.scripting@11.0.10
java.se@11.0.10
...
```
To create JRE with the minimal modules:
```
$ jlink --output jre --add-modules java.base
$ ls jre/bin
java  keytool
$ du -s -h jre
48M     jre
```
To create JRE with all Java SE modules:
```
$ jlink --output jre --add-modules java.se
$ ls jre/bin
java  jrunscript  keytool  rmid  rmiregistry
$ du -s -h jre
94M     jre
$ jre/bin/java --list-modules
java.base@11.0.10
java.compiler@11.0.10
java.datatransfer@11.0.10
java.desktop@11.0.10
java.instrument@11.0.10
java.logging@11.0.10
java.management@11.0.10
java.management.rmi@11.0.10
java.naming@11.0.10
java.net.http@11.0.10
java.prefs@11.0.10
java.rmi@11.0.10
java.scripting@11.0.10
java.se@11.0.10
java.security.jgss@11.0.10
java.security.sasl@11.0.10
java.sql@11.0.10
java.sql.rowset@11.0.10
java.transaction.xa@11.0.10
java.xml@11.0.10
java.xml.crypto@11.0.10
```
To get a list of modules your application depends on (you might need this if the Java SE modules are not enough):
```
$ jdeps vectorgraphics.jar
vectorgraphics.jar -> java.base
vectorgraphics.jar -> java.desktop
vectorgraphics.jar -> java.logging
   org.freehep.graphics2d                             -> java.awt                                           java.desktop
   org.freehep.graphics2d                             -> java.awt.color                                     java.desktop
   org.freehep.graphics2d                             -> java.awt.font                                      java.desktop
   org.freehep.graphics2d                             -> java.awt.geom                                      java.desktop
   org.freehep.graphics2d                             -> java.awt.image                                     java.desktop
   org.freehep.graphics2d                             -> java.awt.image.renderable                          java.desktop
   org.freehep.graphics2d                             -> java.awt.print                                     java.desktop
   org.freehep.graphics2d                             -> java.io                                            java.base
   org.freehep.graphics2d                             -> java.lang                                          java.base
...
```

**Comment by pefeimagic** [comment]
Hi Evgeny

I need the JRE for Windows x86 and x64, Linux x64 and macOS x64.

Thanks
Peter

**Comment by Constantin07** [comment]
HI @eastig I need JRE for Alpine Linux.

**Comment by andrei-bamboi** [comment]
@eastig We want to have separate images for building and separate for running. What your solution says it to adjust current jdk image to include only modules required for an JRE. could be an idea. :) thanks

**Comment by eastig** [comment]
@Constantin07, @pefeimagic Thank you for the information.
We need to figure out what JRE means for Corretto 11 users. Depending on selected modules there can be different versions of JREs.
The current problem is that  OpenJDK 11 source repository does not define a building process of JREs.

**Comment by kennu** [comment]
Hello! We need Corretto 11 to run [Greengrass V2](https://github.com/aws-greengrass/aws-greengrass-nucleus) on Linux on embedded IoT devices. It is bundled with the [meta-aws recipe](https://github.com/aws/meta-aws/tree/master/recipes-iot/aws-iot-greengrass). The problem is that Corretto requires a lot of disk space which is very limited on the devices. We would like to be able to create a minimal installation of Corretto that covers only what Greengrass V2 requires.

We could even do it with a custom solution, if we can just delete some files of the Corretto distribution and still make it work. But I'm not sure how to do that in a supported way.

**Comment by davecurrie** [comment]
@kennu Do you need something even thinner than the jlink solution @eastig provided above?

**Comment by kennu** [comment]
Hey I didn't notice that solution, thanks @davecurrie. We'll try it.

Ultimately it would be nice if Greengrass could bundle with a lightweight prebuilt Corretto distribution specifically tailored for IoT devices. I've been chatting recently with @MikeDombo and @rpcme about Greengrass V2, maybe they have thoughts on the topic..

Right now it's 300MB extra space used on devices that only have a few GB available. (Java was added as a new requirement to Greengrass V2). It's causing some headaches and reducing space available for the actual edge applications.

**Comment by hyperxpro** [comment]
+1 for JRE for all platforms where Corretto JDK is supported.

**Comment by rpcme** [comment]
@kennu somehow I missed your previous comment. Apologies for that. Did you get far in this process?  There is a plan to build corretto from scratch from meta-aws.  This will give us the opportunity to tune for specific distribution types. However, the drawback is (from my understanding) is the binary distribution goes through certification testing whereas the corretto-from-source could not (reasonably) be [compatibility tested](https://openjdk.java.net/groups/conformance/JckAccess/).  If I'm wrong perhaps the corretto team can correct me here.  I am also unsure what the cherry picking of files from the JDK would do to the compatibility status. 

But fundamentally yes I agree it would be helpful to have different distributions.

**Comment by kennu** [comment]
Sounds good @rpcme. I have progressed slowly in my own project (need to migrate a lot of existing Greengrass V1 apps, convert them to use IPC instead of MQTT, and work on related infrastructure simultaneously), so I haven't gotten around to neatly packaging Greengrass V2 and Corretto in our Yocto build yet - just making quick and dirty fixes here and there until everything seems to be working. I am taking a summer break and continuing in August.

Btw, I had to add swap space to our development devices because 512MB quickly runs out with Java. It would be very helpful if AWS-provided packages would optimize also memory space in addition to disk space. With Greengrass V1 and Python this wasn't an issue..

**Comment by rpcme** [comment]
Sounds good @kennu - I am currently on leave through July 12 (of course, still working on open source stuff) so hopefully we can converge paths when you get back.

I will drop a note to folks internally to let them know what you are seeking.

**Comment by davecurrie** [comment]
@rpcme You are right that TCK testing needs to be organized in order to certify any distribution of Corretto from us. Let's discuss when you are back in the office.

**Comment by landsman** [comment]
any progress? I would love to have JRE for 17, 20 as well 🙏

**Comment by miskr-instructure** [comment]
> @Constantin07, @pefeimagic Thank you for the information. We need to figure out what JRE means for Corretto 11 users. Depending on selected modules there can be different versions of JREs. The current problem is that OpenJDK 11 source repository does not define a building process of JREs.

Why not just copy what temurin builds do? I don't think anybody complains there and they offer JRE's for every JDK image. 

https://github.com/adoptium/containers/blob/07677395574f5d3462c3b6fdf5f6c4a0a350b683/21/jre/alpine/Dockerfile

+Headless JRE would also be nice. Servers almost never need the GUI libraries and dependencies.
